towerlib/towerlib.py

```diff
@@ -1908,7 +1908,13 @@ def create_job_template(self,  # pylint: disable=too-many-arguments, too-many-lo
         response = self.session.post(url, json=payload)
         if not response.ok:
             self._logger.error('Error creating job template, response was: "%s"', response.text)
-        return JobTemplate(self, response.json()) if response.ok else None
+            return None
+        job_template = JobTemplate(self, response.json())
+        if credential_id:
+            credential_added = job_template.add_credential_by_id(credential_id)
```

Comment:
```add_credential_by_id``` returns a boolean and does not expose a ```.ok``` atrribute. That is something very specific to the requests library. You would need to check for the return value of the call and not a .ok attribute.

Reply:
Implemented in the new commit

Reply:
The temporary variable ```credential_added``` is not providing value in this case and actually hurts readability. You can just check the return of the method directly.

Reply:
Check, removed the temporary variable in the new commit